adapters/amqp-vertx/src/main/java/org/eclipse/hono/adapter/amqp/AmqpContext.java

```diff
@@ -141,16 +136,19 @@ boolean isDeviceAuthenticated() {
      * case of a <em>ClientErrorException</em>. In the REJECTED case, the supplied exception will provide
      * the error condition value and description as reason for rejection.
      *
-     * @param error The service invocation exception.
+     * @param t The service invocation exception.
      * @throws NullPointerException if error is {@code null}.
      */
-    void handleFailure(final ServiceInvocationException error) {
-        Objects.requireNonNull(error);
-        if (ServerErrorException.class.isInstance(error)) {
-            ProtonHelper.released(delivery, true);
-        } else {
-            final ErrorCondition condition = getErrorCondition(error);
-            MessageHelper.rejected(delivery, condition);
+    void handleFailure(final Throwable t) {
+        Objects.requireNonNull(t);
+        if (ServiceInvocationException.class.isInstance(t)) {
+            final ServiceInvocationException error = (ServiceInvocationException) t;
+            if (ServerErrorException.class.isInstance(error)) {
+                ProtonHelper.released(delivery, true);
+            } else {
+                final ErrorCondition condition = getErrorCondition(error);
+                MessageHelper.rejected(delivery, condition);
+            }
```

Comment:
what if it's not a `ServiceInvocationException`? Don't we need to do anything in that case, e.g. reject the message?